awx/main/management/commands/run_heartbeet.py

```diff
@@ -50,8 +52,15 @@ def construct_payload(self, action='online'):
         }
         return json.dumps(payload)
 
+    def notify_listener_and_exit(self, *args):
+        with pg_bus_conn(new_connection=False) as conn:
+            conn.notify('web_heartbeet', self.construct_payload(action='offline'))
+        sys.exit(1)
+
     def do_hearbeat_loop(self):
         with pg_bus_conn(new_connection=True) as conn:
+            signal.signal(signal.SIGTERM, self.notify_listener_and_exit)
+            signal.signal(signal.SIGINT, self.notify_listener_and_exit)
             while True:
```

Comment:
Not related to this PR, but worth pondering-

It might make more sense for scaling to move this `while` up to the top of the method, out of the `with` block.

The difference is: Right now, we hold on to a connection to the db, in between heartbeat messages. We _could_ just connect, send the message, and disconnect each time. This frees up a persistent (mostly unused) connection to the db.